File: week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    // Validation
    while (height > 8 || height < 1);

    for (int i = 0; i < height; i++)
    {
        // keep track of current amount of spaces to add
        int spaces_to_add = height - i;
        spaces_to_add--;
        for (int j = 0; j < height; j++)
        {
            if (spaces_to_add != 0)
            {
                printf(" ");
                spaces_to_add--;
            }
            else
            {
                printf("#");
            }
        }
        // Print a new line after each row
        printf("\n");
    }
}

File: week6/sentimental-mario-less/mario.py
from cs50 import get_int

while True:
    height = get_int("Height: ")
    if height > 0 and height < 9:
        break


height += 1
for i in range(1, height):
    spaces = height - i - 1
    for j in range(spaces):
        print(" ", end="")
    print("#" * i)
